src/encoded/types/biosample.py

```diff
@@ -685,25 +685,20 @@ def generate_summary_dictionary(
         treatments_list = []
         for treatmentObject in treatment_objects_list:
             to_add = ''
-
-            if experiment_flag is True:
-                if 'treatment_term_name' in treatmentObject:
-                    to_add = treatmentObject['treatment_term_name'] + ' '
-            else:
-                if 'amount' in treatmentObject and \
-                    'amount_units' in treatmentObject:
-                    to_add += str(treatmentObject['amount']) + ' ' + \
-                        treatmentObject['amount_units'] + ' '
-                if 'treatment_term_name' in treatmentObject:
-                    to_add += treatmentObject['treatment_term_name'] + ' '
-                if 'duration' in treatmentObject and \
-                    'duration_units' in treatmentObject:
-                    if treatmentObject['duration_units'][-1] == 's':
-                        to_add += 'for ' + str(treatmentObject['duration']) + ' ' + \
-                            treatmentObject['duration_units']
-                    else:
-                        to_add += 'for ' + str(treatmentObject['duration']) + ' ' + \
-                            treatmentObject['duration_units'] + 's'
+            if 'amount' in treatmentObject and \
```

Comment:
Since we're changing this whole block let's clean up the code.  

Below we put all the dict values into variables that default to empty string, '', if they key doesn't exist in the dict.  Then do the conditions while formatting the string, where the empty strings will evaluate to false.
```
    amt = str(treatment_object.get('amount', ''))
    amt_units = treatment_object.get('amount_units', '')
    treatment_term_name = treatment_object.get('treatment_term_name', '')
    dur = str(treatment_object.get('duration', ''))
    dur_units = treatment_object.get('duration_units', '')
    if dur_units and dur_units[-1] != 's':
        dur_units += 's'
    to_add = "{}{}{}".format(
        (amt + ' ' + amt_units + ' ' if amt and amt_units else ''),
        (treatment_term_name + ' ' if treatment_term_name else ''),
        ('for ' + dur + ' ' + dur_units if dur and dur_units else '')
    )
```

Reply:
done